lifting_veil/agents/dqn_agent_new.py

```diff
@@ -270,10 +274,10 @@ def __init__(self,
 
     def _build_networks_and_optimizer(self):
         self._rng, rng = jax.random.split(self._rng)
-        online_network_params = self.network_def.init(rng, x=self.state, rng=self._rng)
-        optimizer_def = dqn_agent.create_optimizer(self._optimizer_name)
-        self.optimizer = optimizer_def.create(online_network_params)
-        self.target_network_params = copy.deepcopy(online_network_params)
+        self.online_params = self.network_def.init(rng, x=self.state, rng=self._rng)
+        self.optimizer = create_optimizer(self._optimizer_name)
+        self.optimizer_state = self.optimizer.init(self.online_params)
+        self.target_network_params = copy.deepcopy(self.online_params)
```

Comment:
you don't actually need the `copy.deepcopy` anymore

Reply:
You are still using it:
https://github.com/google/dopamine/blob/dabc84b1013b7ced981f189b2bb161e7fa94d31a/dopamine/jax/agents/dqn/dqn_agent.py#L328

Should I remove it?

Reply:
yeah, you can remove it. i should also remove it from the dopamine code.

Reply:
reminder that you can remove `copy.deepcopy`